### PR TITLE
Adds batch/event shape test and fix fialure

### PR DESCRIPTION
**Pull request by feynmanliang**

### CODE REVIEW

File: flowtorch/bijectors/affine_autoregressive.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 
 class AffineAutoregressive(flowtorch.Bijector):
-    event_dim = 1
+    event_dim = 0
```
Comment (feynmanliang) [inline flowtorch/bijectors/affine_autoregressive.py:14]: The batch_dim erroneously gets reinterpreted into an event_dim when this is 1

Comment (stefanwebb) [reply flowtorch/bijectors/affine_autoregressive.py:14]: Ah, shouldn't this be by design? The `AffineAutoregressive` makes the final dimension of a random variable dependent, so will introduce correlations if the base distribution's final dimension is a batch dim

Comment (feynmanliang) [reply flowtorch/bijectors/affine_autoregressive.py:14]: Should `AffineAutoregressive` (or any `Bijector`) continue across batch dimension? Across batch dimension, samples are independent (https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/Distribution#shapes_2) so I'm inclined to say a completely separate bijector (which cannot use autoregressive predictors from the previous batch instance) should be instantiated for a total of `prod(batch_shape)` bijectors.

This is similar to sample_dim, except there we re-use the same `Bijector` due to IID whereas here samples across batch_dim are not identically distributed.

File: tests/test_bijector.py [hunk 2]
```diff
@@ -12,6 +15,26 @@ def test_bijector_constructor():
     assert b is not None
 
 
+@pytest.mark.parametrize(
+    "base_dist",
+    [
+        dist.Normal(0.0, 1.0),
+        dist.Normal(torch.tensor(0.0), torch.tensor(1.0)),
+        dist.Normal(torch.tensor([0.0, 0.0]), torch.tensor([1.0, 1.0])),
+        dist.Independent(
```
Comment (feynmanliang) [inline tests/test_bijector.py:24]: Regardless of semantics, do we think it's reasonable to expect a test that looks like this to pass?

File: tests/test_bijector.py [hunk 3]
```diff
@@ -12,6 +15,26 @@ def test_bijector_constructor():
     assert b is not None
 
 
+@pytest.mark.parametrize(
+    "base_dist",
+    [
+        dist.Normal(0.0, 1.0),
+        dist.Normal(torch.tensor(0.0), torch.tensor(1.0)),
+        dist.Normal(torch.tensor([0.0, 0.0]), torch.tensor([1.0, 1.0])),
```
Comment (feynmanliang) [inline tests/test_bijector.py:23]: nit: to be consistent with what I proposed above, this `batch_shape=(2,)` example should actually instantiate two independent bijectors (one for each independent but not identically distributed batch index)

File: tests/test_bijector.py (hunk 2 above)
Comment (stefanwebb) [reply tests/test_bijector.py:24]: Let's discuss this in person! I think it wouldn't make sense that a transformed distribution has to have the same batch/event shape as the base distribution
